crates/iceberg/src/transaction/snapshot.rs

```diff
@@ -172,6 +179,13 @@ impl<'a> SnapshotProduceAction<'a> {
     // Write manifest file for added data files and return the ManifestFile for ManifestList.
     async fn write_added_manifest(&mut self) -> Result<ManifestFile> {
         let added_data_files = std::mem::take(&mut self.added_data_files);
+        if added_data_files.is_empty() {
+            return Err(Error::new(
+                ErrorKind::Unexpected,
```

Comment:
Maybe we can add new error code like `PreconditionFailed` for cases where we have a precondition for this operation but users failed to meet it.

Reply:
Sounds good! I have removed change against input argument, and use `FailedPrecondition` as the error kind for empty data entry.

The error name is borrowed from grpc status code: https://grpc.io/docs/guides/status-codes/#the-full-list-of-status-codes

Reply:
Hi, it's better to align with our existing error kind naming style that put the noun first: `PreconditionFailed`.

Reply:
Ah sorry I didn't realize it, thanks for the heads up! Updated.